robot/web/stream/server.go

```diff
@@ -412,3 +486,97 @@ func (server *Server) removeMissingStreams() {
 		utils.UncheckedError(streamState.Close())
 	}
 }
+
+// refreshVideoSources checks and initializes every possible video source that could be viewed from the robot.
+func (server *Server) refreshVideoSources() {
+	for _, name := range camera.NamesFromRobot(server.robot) {
+		cam, err := camera.FromRobot(server.robot, name)
+		if err != nil {
+			continue
+		}
+		existing, ok := server.videoSources[cam.Name().SDPTrackName()]
+		if ok {
+			existing.Swap(cam)
+			continue
+		}
+		newSwapper := gostream.NewHotSwappableVideoSource(cam)
```

Comment:
This is where the type in go matters right? Is it because camera now fulfills Strem and Next? If so, a good comment would be to figure out how to wrap GetImage in the required type for gostream for our plans.

Reply:
Probably the latter, I think this is the place where we need to decouple the interfaces or have one wrap the other.

Reply:
Yes exactly thinking it will be similar to `NewVideoSourceFromReader` helper we currently have.